tests/integ/sagemaker/jumpstart/conftest.py

```diff
@@ -80,17 +79,18 @@ def _teardown():
 
     inference_component_names = []
     for endpoint_name in endpoint_names:
-        for inference_component in sagemaker_client.list_inference_components(
-            EndpointNameEquals=endpoint_name
-        ).get("InferenceComponents", []):
-            inference_component_names.append(inference_component["InferenceComponentName"])
+        for (
+            inference_component_name
+        ) in sagemaker_client.list_inference_components_associated_with_endpoint(
+            endpoint_name=endpoint_name
+        ):
+            inference_component_names.append(inference_component_name)
 
     # delete inference components for test-suite-tagged endpoints
     for inference_component_name in inference_component_names:
-        sagemaker_client.delete_inference_component(InferenceComponentName=inference_component_name)
-
-    # wait for sagemaker to detect IC has been deleted
-    time.sleep(60 * 5)
```

Comment:
sanity-check: no wait needed here? I know it's a test config so it may be fine, but generally if you calla delete API in a loop, you're bound to hit throttling limits.